app/main/service/user_service.py

```diff
@@ -1 +1,38 @@
-# for user related operations
\ No newline at end of file
+# for user related operations
+
+import uuid
+import datetime
+
+from app.main import db
+from app.main.models.user import User
+
+def save_new_user(data):
+    """Creates a new user"""
+    user = User.query.filter_by(email=data['email']).first()
+    if not user:
+        new_user = User(
+            # generates a random id
+            public_id=str(uuid.uuid4())
```

Comment:
We do not need to generate a random id since we will make these autoincrement and hence they'll be generated automatically. So, I think you should remove all the code required for this.